mindsdb/interfaces/knowledge_base/controller.py

```diff
@@ -112,6 +116,28 @@ def delete_query(self, query: Delete):
         db_handler = self.get_vector_db()
         db_handler.query(query)
 
+    def hybrid_search(
+        self,
+        query: str,
+        keywords: List[str] = None,
+        metadata: Dict[str, str] = None,
+        distance_function=DistanceFunction.COSINE_DISTANCE
+    ) -> pd.DataFrame:
+        query_df = pd.DataFrame.from_records([{TableField.CONTENT.value: query}])
+        embeddings_df = self._df_to_embeddings(query_df)
```

Comment:
it doesn't look like `self._df_to_embeddings` method actually guaranteed to return a non empty dataframe. 
maybe it could be good to add a check for that